app/services/the_tvdb_service.rb

```diff
@@ -0,0 +1,95 @@
+class TheTvdbService
+  BASE_URL = 'https://api.thetvdb.com'.freeze
+  API_KEY = ENV['THE_TVDB_API_KEY'].freeze
+
+  def initialize
+    @token = api_token
+  end
+
+  def nightly_import!
+  end
+
+  def weekly_import!
+    return if missing_data.empty?
+
+    missing_data['Anime'].each do |media_ids|
+      media_id = media_ids[0]
+
+      # check if the external_sites exist.
+      series_id = media_ids[1].try(:[], 'thetvdb/series')
```

Comment:
You might be better off just doing `next unless media_ids[1]` before these so you don't need to do that `try(:[]` stuff